newsrevenuehub/rev-engine
# Dev 3177 integrations callout details links and details modals

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a

#### What's this PR do?
- Create Mailchimp "View details" modal
- Create Modal Upgrade prompt
- Implement Modal to Mailchimp Integration Card if user has feature flag enabled
- Create new base component "IconList"
- Make IntegrationCardHeader a separate component to be utilized in multiple locations
- Update Link and ModalHeader base components style (small updates)
- Update useConnectMailchimp hook to return feature flag
- Fix broken test 
 
#### Why are we doing this? How does it help us?
Adds a modal with more details about mailchimp connection for users. Modal shows unique information in each stage related to the connection. 

#### How should this be manually tested? Please include detailed step-by-step instructions.
6 test scenarios:
1. User with mailchimp feature flag disabled
   - "View details" should not show in Mailchimp integration card
   - Modal not accessible 
   - Everythin should work as usual
2. User with mailchimp feature flag enabled and Org plan = FREE
   - "View details" should show in Mailchimp integration card
   - Click that button to show Modal 
   - Modal with upgrade prompt should display
   - CTAs: Learn More, Maybe later, Upgrade
3. User with mailchimp feature flag enabled, Org plan = CORE, Mailchimp NOT connected
   - "View details" should show in Mailchimp integration card
   - Click that button to show Modal 
   - Modal with upgrade prompt should not display
   - CTAs: Support, Maybe later, Connect
4. User with mailchimp feature flag enabled, Org plan = PLUS, Mailchimp NOT connected
   - "View details" should show in Mailchimp integration card
   - Click that button to show Modal 
   - Modal with upgrade prompt should not display
   - CTAs: Support, Maybe later, Connect
5. User with mailchimp feature flag enabled, Org plan = CORE, Mailchimp connected
   - "View details" should show in Mailchimp integration card
   - Click that button to show Modal 
   - Modal header should be "Successfully Connected!"
   - CTAs: FAQ, Close, Go to Contributions
   - No bullet list copy as of yet (TODO)
6. User with mailchimp feature flag enabled, Org plan = PLUS, Mailchimp connected
   - "View details" should show in Mailchimp integration card
   - Click that button to show Modal 
   - Modal header should be "Successfully Connected!"
   - CTAs: FAQ, Close, Go to Contributions
   - No bullet list copy as of yet (TODO)

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
Yes

#### Have automated unit tests been added? If not, why?
yes

#### How should this change be communicated to end users?
Mailchimp modal added to help user with more information about what the connection does and how to connect.

#### Are there any smells or added technical debt to note?
no

#### Has this been documented? If so, where?
no

#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-3177

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
no

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
I think not.
